### ISSUE TITLE
Make loss + backward more configurable

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/PiPPy#13]
Currently, when building a `Pipe` object, if the user specifies a `loss_fn`, we automatically insert the loss function as well as the backward pass: https://github.com/jamesr66a/PiPPy/blob/527af1fd8123d35bd81b9fe304a8d0ed29c9fd8d/pippy/IR.py#L314

Potentially a user might want to just insert the loss function. We could make this more configurable
